src/legacy/ui/public/vis/map/kibana_map.js

```diff
@@ -564,6 +564,8 @@ export class KibanaMap extends EventEmitter {
 
     let baseLayer;
     if (settings.baseLayerType === 'wms') {
+      //This is user-input that is rendered with the Leaflet attribution control. Needs to be sanitized.
+      this._baseLayerSettings.options.attribution = _.escape(settings.options.attribution);
```

Comment:
Is there a reason why we don't want to do the escaping regardless of the `settings.baseLayerType`? Is there any way the user could manually modify the Elasticsearch document to get have the baseLayerType be neither `wms` or `tms`?

Reply:
> Is there a reason why we don't want to do the escaping regardless of the settings.baseLayerType?

Yes. When the baselayer-type is `tms` it comes from the `tilemap.*`-config in kibana.yml`. This can be markdown and is sanitized here https://github.com/thomasneirynck/kibana/blob/7bcc7634ec10c3b035d256f4c4bc26258ac53294/src/legacy/ui/public/vis/map/service_settings.js#L38

> Is there any way the user could manually modify the Elasticsearch document to get have the baseLayerType be neither wms or tms?

If it is a `wms`, it is stored in the saved-object for that visualization in the .kibana-index. So that attribution could be "hacked". If it's `tms`, attribution comes from the servers-side map-config, or the Elastic Maps Service manifest which is hosted by Elastic. The actual `wms` and `tms` strings though or part of the logic-flow in the code and cannot be directly modified.